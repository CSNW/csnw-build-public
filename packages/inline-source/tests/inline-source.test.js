const { join } = require('path');
const { Files } = require('@csnw-build/pipeline');

process.env.PUBLIC_URL = '/blog/';

const inlineSource = require('../')({});
const cwd = join(__dirname, 'fixtures');

test('pipeline', async () => {
  const process = await inlineSource.pipeline({
    cwd,
    cache: false
  });
  const files = await Files.from('**/*', { cwd });

  const result = await process(files);
  expect(result).toMatchSnapshot();
});
