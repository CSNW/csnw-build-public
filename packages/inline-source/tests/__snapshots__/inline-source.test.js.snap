// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pipeline 1`] = `
Object {
  "index.html": Object {
    "baseDir": "C:/dev/csnw-build/packages/inline-source/tests/fixtures",
    "contents": "<!doctype html>
<html>
  <head>
    <title>inline-source</title>
  </head>
  <body>
    <script>var name=\\"a\\";</script>
    <script src=\\"/blog/main.js\\"></script>
  </body>
</html>",
    "cwd": "C:/dev/csnw-build/packages/inline-source/tests/fixtures",
    "path": "C:/dev/csnw-build/packages/inline-source/tests/fixtures/index.html",
    "sourceMap": null,
  },
  "main.js": Object {
    "baseDir": "C:/dev/csnw-build/packages/inline-source/tests/fixtures",
    "contents": "var name = 'main';
",
    "cwd": "C:/dev/csnw-build/packages/inline-source/tests/fixtures",
    "path": "C:/dev/csnw-build/packages/inline-source/tests/fixtures/main.js",
    "sourceMap": null,
  },
}
`;
